### DISCUSSION TITLE
Blazor Deprecate Warning in 2024.3.0 release

### DISCUSSION THREAD

**Discussion by fingers10** [QuestPDF/QuestPDF#822] [General]
I'm surprised to see the following warning in the [official docs](https://www.questpdf.com/going-production/platform-specific-dependencies.html#support-for-custom-environments-blazor-webassembly)

> The WebAssembly support is likely to be deprecated in the 2024.3.0 release. Please consider using the server-side Blazor or using the 2023.12.X release.

Please can the team share the rationale behind this? This is a wonderful package and stopping the support in future release is disappointing.

**Comment by user-named-void** [comment]
Please note that the team is mostly one person.
In 2024.3.0 the implementation of Skia (which does the actual pdf creation) has completely changed, instead of relying on SkiaSharp, which had some challenges, Marcin build his own Skia library (more info can be found [here](https://github.com/QuestPDF/QuestPDF/discussions/622#discussioncomment-8825050)). Because of this it is probably too much work to support all edge cases and possible implementations at this time.
You can read more about this in [his recent post](https://github.com/QuestPDF/QuestPDF/discussions/622#discussioncomment-8825050) about the latest beta which mentions the WebAssembly.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

To clarify the rationale behind this decision — QuestPDF's text rendering engine relies on several megabytes of Unicode formatting data essential for proper text shaping, bi-directional text support, and complex script handling. Requiring end users to download this data in a browser environment introduces significant overhead and fundamentally undermines the performance benefits that PDF generation should offer. In practice, server-side generation (including server-side Blazor) is not just the recommended approach — it is the only approach that delivers the performance and reliability users should expect.

Should community demand for WebAssembly support grow significantly, we may reconsider this in the future. However, for the foreseeable future, this remains outside the scope of active development.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
